crates/runtime/src/component/view.rs

```diff
@@ -14,29 +14,84 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+use app::App;
 use datafusion::sql::TableReference;
 use serde_json::Value;
 use snafu::prelude::*;
 use spicepod::component::view as spicepod_view;
-use std::{collections::HashMap, fs};
+use std::{collections::HashMap, fs, sync::Arc};
+
+use crate::{Runtime, dataaccelerator::AccelerationSource};
 
 use super::{
-    dataset::{Dataset, acceleration},
+    dataset::{
+        Dataset,
+        acceleration::{self, Acceleration},
+    },
     validate_identifier,
 };
 use spicepod::semantic::Column;
 
 /// [`View`] is the internal representation of the [`spicepod_view::View`] spicepod component.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Clone)]
```

Comment:
This is relatively expensive to Clone - let's either make it inexpensive (i.e. wrap internal components with Arc) or work with an Arc<View> directly

Reply:
Thank you - updated to operate by Arc<View> to avoid cloning